### ISSUE TITLE
Xorg memory leak

### ISSUE THREAD

**Issue by th3str4p** [regolith-linux/regolith-desktop#453]
- [x ] I've [run a search on the existing bug reports](https://help.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests) to make sure I'm not filing a duplicate (you can use the search box at the top left corner)
- [ x] I've read [the Regolith documentation](https://regolith-linux.org/docs/https://regolith-linux.org/docs/) and didn't find any relevant topic
- [ x] I tried [asking on Slack](https://regolith-linux.org/interact/) (especially for configuration related issues)

**Describe the bug**
Xorg will slowly and steadily climb in RAM usage no matter what programs are or aren't open.  The only thing that frees the RAM back up is completely restarting the Xorg session.

**To Reproduce**
1.  Boot into Regolith 1.4.1 session on Ubuntu 20.04.
2.  Watch the Xorg RAM usage steadily climb.

**Expected behavior**
Xorg shouldn't be eating RAM like this.

**Screenshots**
N/A

**Configuration file(s)**
I've only done minor configuration such as adding some i3xrocks blocks to my bar, enabling the system tray and moving the bar to the top of the screen (essentially just things that are outlined in the how-to documentation).

**Installation Details**
 - Regolith Install Type: PPA
 - Regolith Version: 1.4.1
 - PPA url: ppa:regolith-linux/release
 - Host OS (for PPA): Ubuntu 20.04

**Additional context**
On the recommendation of @kgilmer I tried rebooting into a standard GNOME Xorg session and let the machine sit overnight.  RAM usage didn't really climb (went from ~27MB at boot to ~31MB by the next morning).  Typically when logged into Regolith overnight it will go from ~27MB to ~1.5GB overnight.  So this definitely seems to be a bug that is limited to the Regolith Xorg session.

Thanks in advance for helping me track down and fix this issue!

**Comment by th3str4p** [comment]
Just a quick update, I'm at ~9 hours since rebooting into Regolith and posting this bug report, and Xorg has grown to 2.3GB of RAM usage.

![2020-06-18_17-22](https://user-images.githubusercontent.com/25623116/85084002-59e3ee80-b188-11ea-99a5-adcfeb26d1ef.png)

**Comment by kgilmer** [comment]
The first thing I'd suggest is removing the status indicators and see if that's it.  You can learn how to do this here: https://regolith-linux.org/docs/howto/add-remove-blocklets/

**Comment by th3str4p** [comment]
Thanks @kgilmer!  I removed every single indicator so that the only thing left is the notification indicator and the system tray.  That seems to have potentially fixed the issue or at least dramatically slowed down the leak.  It seemed to climb from ~27MB up to ~49MB and has been sitting around that mark since.

Obviously this isn't ideal as it means I don't have even a clock in my bar, so I'm going to start adding the indicators back in one by one and see if I can narrow down which one of them is the worst offender.  I'll keep you posted.

**Comment by th3str4p** [comment]
Just a further quick update:  It definitely seems to have slowed the leak but not stopped it entirely.  It has kept climbing up to ~96MB of RAM usage now.

**Comment by th3str4p** [comment]
Still slowly creeping up at ~158MB now.

**Comment by kgilmer** [comment]
Can you try disabling i3xrocks althogether?  You can do this by setting the Xres override for i3xrocks, the key is `i3-wm.bar.status_command` and you can set the value to ":".  You can learn more here: https://regolith-linux.org/docs/howto/override-xres/

cat "i3-wm.bar.status_command: :" >> ~/.config/regolith/Xresource

then log back in.

**Comment by th3str4p** [comment]
Ok, a further update.  I completely disabled i3xrocks and replaced the i3 bar with Polybar.  I also removed Rofication and replaced it with dunst.

This appears to have slowed the memory leak even further.  However, still after ~15 hours the RAM usage of Xorg has crept up to ~316MB.  Much better than hitting 1.5-2GB within that same period previously, but still not completely resolved.

**Comment by muktupavels** [comment]
Compositor and/or WM?

**Comment by kgilmer** [comment]
@th3str4p , @muktupavels has a good idea.  Let's turn off the compositor and see what happens.  To do that; `sudo apt install regolith-compositor-none`.  More info here: https://regolith-linux.org/docs/customize/compositors/

**Comment by moritzheiber** [comment]
Which process specifically is consuming more RAM? Is it the actual `Xorg` process? Or something else?

**Comment by krisman** [comment]
I stumbled upon this PR while googling for the same issue.  Pretty sure it is not specific to Regolith specific packages. I got a fresh Debian install with the distro's i3 and i3blocks packages slowly eating my RAM in the same fashion.  similar to this PR,  system credits the RAM comsumption to xorg.  This started around one month ago in the Debian package,  if that helps to bisect the package version that caused it.

Just fyi, as I'm report it to Debian.
